File: checkotp.php
<?php
function checkOTP($pin,$otp,$initsecret)
{
 $maxperiod = 3*60; // in seconds = +/- 3 minutes
 $time=gmdate("U");
 echo $time;
 for($i = $time - $maxperiod; $i <= $time + $maxperiod; $i++)
 {
    $md5 = substr(md5(substr($i,0,-1).$initsecret.$pin),0,6);
    if($otp == $md5) return(true);
 }
return(false);
}

echo checkOTP('3216','9f9fc1d','d41d8cd98f0b24e9');
?>

File: checkotp.py
import time
import hashlib

def checkMOTP(pin, otp, initsecret):
	maxperiod = 3*60
	etime = int(time.time())
	i = etime - maxperiod
	while i <= etime + maxperiod:
		n = hashlib.md5()
		n.update(str(i)[0:-1] + str(initsecret) + str(pin))
		md5 = n.hexdigest()[0:6]
		if otp == md5:
			return True
		i += 1
	return False

print checkMOTP(3216, 'a058f7', 'd41d8cd98f0b24e9')
